src/parse/trainer.py

```diff
@@ -146,9 +149,21 @@ def _produce_trainer_stages(data: dict) -> TrainerSteps:
 
     model = get_model(stages[0]['model'] | { "x_dim" : x_dim }, x_dim, y_dim)
     optimizer = get_optimizer(model, stages[0]['optim'])
-    if 'model_weights' in data and 'optim_state' in data:
+
+    resume = 'model_weights' in data and not 'base_model' in stages[0]['model']
+    lora = 'base_model' in stages[0]['model'] and type(model) == MODELS['lora']
```

Comment:
this line worries me, but I need it in order to make sure that `model.setup_peft_model (...)` exists.

Reply:
Perhaps a nesting structure would help:
```
{
    "optimizer_state": ...,
    "model_weights": <ContextModel weights>,
    "fine_tune": {
        "model_weights": <Adapter 1 weights>,
        "fine_tune": {
            "model_weights": <Adapter 2 weights>,
        }
    }
}
```